pkg/common/peertracker/credentials.go

```diff
@@ -38,7 +38,7 @@ func (c *grpcCredentials) Info() credentials.ProtocolInfo {
 }
 
 func (c *grpcCredentials) Clone() credentials.TransportCredentials {
-	return &(*c)
+	return &(*c) // nolint: staticcheck // cloning
```

Comment:
from what I understand from the golint check here, it is saying that `&(*c)` may be replaced with just `c` and therefore we should probably be explicit here:
```suggestion
	copy := *c
	return &copy
```